### ISSUE TITLE
geometry restructuring

### ISSUE THREAD

**Issue by jan-kaspar** [CTPPS/cmssw#12]
The purpose of this discussion is two-fold:
   * find a fast acceptable solution before legacy re-reco of 2016 data, thus timescale in days
   * find a "final" optimal solution, in summer we may have students to do what needed.

First, let me explain the current code structure.

Since the Roman Pot detectors are movable, their alignment is of primary importance. For the alignment studies we have found convenient to have two geometry descriptions:
  * `VeryForwardMisalignedGeometryRecord`: used for simulations including mis-alignments
  * `VeryForwardRealGeometryRecord`: used for reconstruction and alignment-determination algorithms.

Later, a third geometry was added: `VeryForwardMeasuredGeometryRecord`. The reason is mostly historic and I believe it can be removed. Here are the current and simplified schemes:
  * [geometry_building_current.pdf](https://github.com/CTPPS/cmssw/files/770617/geometry_building_current.pdf)
  * [geometry_building_simplified2.pdf](https://github.com/CTPPS/cmssw/files/770689/geometry_building_simplified2.pdf) NB: no manipulations with `DDCompactView`.

The only issue with the simplified solution could be related to Geant4-based simulations - this was the case when we implemented the geometry code (years ago). We needed (and most likely still need) to run simulations with the same alignments as in reconstruction. On the other hand, the Geant4 simulation (OscarXXX) accepted only geometry in `DDCompactView` format. That is why we had to put in place the `MeasuredGeometryProducer`. Would you know whether it is possible to pass any sort of "expanded" geometry to Geant4 now?

For the short-term solution, I propose to follow the simplified scheme  [geometry_building_simplified2.pdf](https://github.com/CTPPS/cmssw/files/770689/geometry_building_simplified2.pdf). It is only simplification of the current code, it avoids manipulations with `DDCompactView` and it avoids the non-thread-safe `MeasuredGeometryProducer`. The only possible downside is that we might not be able to run simulations with alignments - which is not a problem now. Thoughts?

In longer term we need Geant4 simulations with alignments. Any recommendations?

**Comment by slava77** [comment]
@ianna @davidlange6 @dan131riley
this is a rebase of earlier xerces and DD and related discussions

**Comment by jan-kaspar** [comment]
The short term solution has been implemented in #17535.

**Comment by ianna** [comment]
Here are a couple of comments from a first look at the geometry description.

1. The following files can be safely dropped from the description. I'll submit a PR later today.
```
'Geometry/CMSCommonData/data/beampipe/2015/v1/beampipe.xml', 
'Geometry/CMSCommonData/data/cmsBeam.xml', 
'Geometry/CMSCommonData/data/mgnt.xml',
```
2. Geometry hierarchy and/or volume size should be changed. All RP* and CTPPS volumes are children of CMSE, but they are outside of it, or even the World volume: see the screenshot attached. On the other hand, are they actually geometric description of Totem geometry or some parameters which should be applicable to it?
<img width="1323" alt="screen shot 2017-03-14 at 16 01 51" src="https://cloud.githubusercontent.com/assets/1390682/23906908/c1b4f0c8-08cf-11e7-9a42-f947c4ece8f0.png">

**Comment by ianna** [comment]
@jan-kaspar - how often do the alignments change during a run?

**Comment by jan-kaspar** [comment]
> how often do the alignments change during a run? and what is a condition for it?

The answer is two-fold.

In general, we align the RPs wrt. the LHC beams. And it has happened that we could measure beam movements during a fill. In such cases we update the alignment constants every few minutes.

For CTPPS we will most likely not have such a sensitivity and we will produce a set of alignment constants per fill. For the moment, moreover, as we are in an exploratory phase, the alignment is an analysis on its own, producing results with a noticeable delay after data-taking. Consequently, as we might not be in time for including the alignment corrections in (re)-reco, we apply them on physics-analysis level. In principle, the methods have been established and we should go for automatisation, but with changing detector technologies in the RPs, it is difficult to foresee the time scale.

To summarise, in a reasonable future I guess that there will be 1 set of alignment corrections per CMS run. Most often there will be the same corrections for the entire LHC fill.

**Comment by grzanka** [comment]
@ianna Ad 2. I confirm that CTPPS is at location of ~200m from IP. CMSE needs to be extended to include these detectors. For that purpose we usually include 
```Geometry/CMSCommonData/data/extend/cmsextent.xml#L11`` which has CMSZ1  value set to 450.0*m

Is it the same thing as you are planning to extend this volume ?

**Comment by jan-kaspar** [comment]
Remainder of this discussion moved to https://github.com/CTPPS/cmssw/issues/40.
